lib/go/thrift/simple_server.go

```diff
@@ -234,7 +250,9 @@ func (p *TSimpleServer) Stop() error {
 	}
 	atomic.StoreInt32(&p.closed, 1)
 	p.serverTransport.Interrupt()
+	close(p.stopChan)
 	p.wg.Wait()
+	p.stopChan = make(chan struct{})
```

Comment:
Because is stopped server can serve later, and the closed channel can not be closed again, so we have to make a new channel

Reply:
in that case this should be done in `Serve` instead.

Reply:
I think it is better to init the stopChan in construction, if not TestMultipleStop will panic because of nil chan

What's more, current HEAD allows multi **Serve**, if we make it in **Serve**，the stopChan may change and will never be closed

Reply:
i don't think so
Firstly, the stopChan should be created when the server is created
Secondly, it is **stop** that close the stopChan, so **stop** should also be responsible to recreate the stopChan
After all if we call serve several times before stop, it will create stopChan several times without close it, that is quite strange